packages/asset-swapper/contracts/src/SamplerBase.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2021 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6;
+pragma experimental ABIEncoderV2;
+
+contract SamplerBase
+{
+    /// @dev Stored Sample values for each Sampler to pull
+    uint256[] private SAMPLE_VALUES = new uint256[](0);
+
+    function getSampleValues()
+        public
+        view
+        returns (uint256[] memory sampleValues)
+    {
+        sampleValues = SAMPLE_VALUES;
+    }
```

Comment:
what if we just did this, have all the sampler contracts inherit `SamplerBase`, and simply access `SAMPLE_VALUES` directly so we can reduce gas and bytecode further.

```suggestion
    uint256[] internal SAMPLE_VALUES;
```

Reply:
And also mark all the non-global `sample*From*()` functions private/internal.

Reply:
My intuition says we still need to keep some of these public for the Approximate buys, but I haven't got there yet.